pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -107,3 +108,21 @@ def attach_model_logging_functions(self, model):
         for callback in self.trainer.callbacks:
             callback.log = model.log
             callback.log_dict = model.log_dict
+
+    def attach_model_callbacks(self, model: LightningModule) -> None:
+        """
+        Attaches the callbacks defined in the model.
+        If a callback returned by the model's configure_callback method has the same type as one or several
+        callbacks already present in the trainer callbacks list, it will replace them.
+
+        Args:
+            A model which may or may not define new callbacks in
+            :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_callbacks`.
+        """
+        model_callbacks = model.configure_callbacks()
+        model_callback_types = [type(c) for c in model_callbacks]
+        # remove all callbacks with a type that occurs in model callbacks
+        all_callbacks = [c for c in self.trainer.callbacks if type(c) not in model_callback_types]
```

Comment:
will it make sense to have some warning here?

Reply:
I would say definitely not a warning, because the way it is designed right now it is clearly a feature and the user first has to override the hook (and therefore read the docs) in order for all of this to take effect. 

One could argue maybe for a INFO or DEBUG level log statement

Reply:
INFO sound good

Reply:
I added that info. Let me know if you see any further improvements

Reply:
looks good.